### DISCUSSION TITLE
Firewall rules to allow WAN/internet access from LAN

### DISCUSSION THREAD

**Discussion by eadmaster** [MichaIng/DietPi#6007] [Q&A] [answered]
My [NanoPi R1 board has 2 ethernet ports](http://wiki.friendlyelec.com/wiki/index.php/NanoPi_R1), and i'd like to allow the PC connected to the LAN port to have internet access (the WAN port is connected to my ISP router)

With Operwrt i've created these firewall zones to allow this, i am not sure how to do the same in Dietpi:

![Screenshot_20221224_020240-1](https://user-images.githubusercontent.com/925171/209416264-a10282b1-4eb6-4557-80df-e4f252886f2c.png)

my current iptables rules:
````
(ssh) root@DietPi:~/bin# iptables --list-rules

-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT
-N DOCKER
-N DOCKER-ISOLATION-STAGE-1
-N DOCKER-ISOLATION-STAGE-2
-N DOCKER-USER
-A FORWARD -j DOCKER-USER
-A FORWARD -j DOCKER-ISOLATION-STAGE-1
-A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -o docker0 -j DOCKER
-A FORWARD -i docker0 ! -o docker0 -j ACCEPT
-A FORWARD -i docker0 -o docker0 -j ACCEPT
-A FORWARD -i eth0 -o wlan0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -i wlan0 -o eth0 -j ACCEPT
-A DOCKER-ISOLATION-STAGE-1 -i docker0 ! -o docker0 -j DOCKER-ISOLATION-STAGE-2
-A DOCKER-ISOLATION-STAGE-1 -j RETURN
-A DOCKER-ISOLATION-STAGE-2 -o docker0 -j DROP
-A DOCKER-ISOLATION-STAGE-2 -j RETURN
-A DOCKER-USER -j RETURN
````

**Comment by Joulinar** [comment]
Do you need to configure a different IP range on the LAN interface? Or should It be same network segment?

**Reply by eadmaster** [reply]
i have no particular needs for this, the easier setup will do.

**Reply by Joulinar** [reply]
Probably some bridged network interfaces would need to be created.

**Comment by MichaIng** [comment, marked as answer]
These should do:
```sh
iptables -t nat -A POSTROUTING -s 192.168.42.0/24 -o eth0 -j MASQUERADE
iptables -A FORWARD -i eth0 -o eth1 -m state --state RELATED,ESTABLISHED -j ACCEPT
iptables -A FORWARD -i eth1 -o eth0 -j ACCEPT
ip6tables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
ip6tables -A FORWARD -i eth0 -o eth1 -m state --state RELATED,ESTABLISHED -j ACCEPT
ip6tables -A FORWARD -i eth1 -o eth0 -j ACCEPT
```
This expects the WAN port to be `eth0` and the LAN port to be `eth1`. And a DHCP server or all static IP configs at `192.168.42.0/24` subnet. Otherwise adjust as needed.
